### ISSUE TITLE
[NETSDKE2E]Garbage issue happens in new terminal logger output in CLI on Loc OS.

### ISSUE THREAD

**Issue by vdanche** [dotnet/msbuild#9674]
### Issue Description

Garbage issue happens in new terminal logger output in CLI on Loc OS.

### Steps to Reproduce

Steps to Reproduce
Affect build:
.NET 9.0.100-alpha.1.24072.8 from [dotnet/installer: .NET SDK Installer (github.com)](https://github.com/dotnet/installer)
Affected Product Language:
All loc OS

1.In CLI, dotnet new mstest -o mymstest本地化
2.add below code in UnitTest1.cs

        [TestMethod]
        public void TestMethod1()
        {
            Assert.Inconclusive();
        }

        [TestMethod]
        public void TestMethod2()
        {
            Assert.AreEqual(2, 11);

        }
3.dotnet build
4.dotnet  test

**Expected Result:**
the log can be displayed well.
**Actual Result:**
Garbage issue happen in logs.
Chinese OS:
![image](https://github.com/dotnet/msbuild/assets/65637796/d91f2e26-20e2-42bc-97fb-437be70bddd8)

Turkish:
![image](https://github.com/dotnet/msbuild/assets/65637796/1e6741d7-ed19-4f5b-b310-b73216343ecd)

EN(not repro):
![image](https://github.com/dotnet/msbuild/assets/65637796/d3b2bcc3-5930-4529-a246-21d5edff99eb)

dotnet --info:
.NET SDK:
Version:           9.0.100-alpha.1.24072.8
Commit:            ac061bfeca
Workload version:  9.0.100-manifests.fd1572dc
 
运行时环境:
OS Name:     Windows
OS Version:  10.0.22621
OS Platform: Windows
RID:         win-x64
Base Path:   C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24072.8\
 
已安装 .NET 工作负载:
没有要显示的已安装工作负载。
 
Host:
  Version:      9.0.0-alpha.1.24071.3
  Architecture: x64
  Commit:       8accd8073a
 
.NET SDKs installed:
  9.0.100-alpha.1.24072.8 [C:\Program Files\dotnet\sdk]
 
.NET runtimes installed:
  Microsoft.AspNetCore.App 9.0.0-alpha.1.24069.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 9.0.0-alpha.1.24071.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 9.0.0-alpha.1.24071.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
 
Other architectures found:
  None
 
Environment variables:
  Not set
 
global.json file:
  Not found
 
Learn more:
https://aka.ms/dotnet/info
 
Download .NET:
https://aka.ms/dotnet/download

**Comment by rainersigwald** [comment]
@nohwnd, this is the new test logging experience, right?

**Comment by nohwnd** [comment]
Yes. Will investigate tomorrow. I looks like encoding issue.

Do you have an idea of a possible cause? I don’t do much in the logger, just write the test names and results to standard output and consume those lines in class VstestTask2 : ToolTask. I replace \n and \r .

**Comment by rainersigwald** [comment]
@nohwnd It is _completely plausible_ that you're not doing anything wrong and the encoding problem is happening after your perfectly-ok log messages get to the logger. But I definitely wanted to make sure you were aware of the end-to-end problem.

**Comment by nohwnd** [comment]
I can repro on my computer, that is using english: 

```
namespace mstest95;

[TestClass]
public class UnitTest1ğ
{
    [TestMethod]
    public void TestMethodğ1()
    {
        Assert.AreEqual(2, 1);
    }
}
```


![image](https://github.com/dotnet/msbuild/assets/5735905/299d728b-c07b-412e-88a3-dd2461925d33)

Investigating.

**Comment by nohwnd** [comment]
This is indeed encoding issue, seems that the task is not forcing utf8 by default, which we do force in the console unless user opts out. (using this set of characters to test, it works when UTF8 is set "ğğğ𦮙我們剛才從𓋴𓅓𓏏𓇏𓇌𓀀")

**Comment by nohwnd** [comment]
Will be fixed in the linked PR, 
![image](https://github.com/dotnet/msbuild/assets/5735905/5772824b-d2f5-40bc-966e-e0360005413c)

Formatting is different because of different SDK  version, but we reach MSBuild with the correct string, so this will fix the new output shown above as well.

**Comment by vdanche** [comment]
@nohwnd It will be fixed on net9 preview 2 instead of net9 preview 1, right? we checked on 9.0.100-preview.2.24080.8 and still repro, will keep tracking.
![image](https://github.com/dotnet/msbuild/assets/65637796/27737b16-ca76-44cd-9dc4-b0d2aebdc677)

**Comment by nohwnd** [comment]
Fix was merged yesterday to net9 main, and few hours ago into net9-preview1. So it should be in both.

**Comment by vdanche** [comment]
> Fix was merged yesterday to net9 main, and few hours ago into net9-preview1. So it should be in both.

Ok, still repro on today's net9 preview 2 and net9 preview 1. we'll check it on tomorrow build.

**Comment by vdanche** [comment]
@nohwnd this issue has been fixed on 9.0.100-preview.1.24101.4 SDK. thanks
![image](https://github.com/dotnet/msbuild/assets/65637796/d7e9e528-6d49-4a2a-bd78-25cf7693903c)
